### PR TITLE
feat: introduce google client

### PR DESCRIPTION
**Pull request by dzehnder**
This PR introduces the GoogleClient, designed to facilitate interactions with Google APIs for retrieving organic search data and listing sites. The class uses AWS Secrets Manager to securely manage Google OAuth credentials, ensuring secure and efficient API requests.

### CODE REVIEW

**Comment by dzehnder** [comment]
relates to https://github.com/adobe/spacecat-shared/pull/219

File: packages/spacecat-shared-google-client/src/index.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    }
+    return new GoogleClient(authClient, context.log);
+  }
+
+  constructor(authClient, log = console) {
+    this.authClient = authClient;
+    this.log = log;
+  }
+
+  async getOrganicSearchData(baseURL, startDate, endDate) {
```
Comment (ekremney) [inline packages/spacecat-shared-google-client/src/index.js:44]: we could add `startRow` and `rowLimit` as params with default values 0 and 10. That way, this API could also provide a way to paginate

File: packages/spacecat-shared-google-client/src/index.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+export default class GoogleClient {
+  static async createFrom(context) {
+    const authClient = new OAuth2Client(
+      context.env.GOOGLE_CLIENT_ID,
+      context.env.GOOGLE_CLIENT_SECRET,
+      context.env.GOOGLE_REDIRECT_URI,
+    );
+
+    authClient.setCredentials({
+      access_token: context.env.ACCESS_TOKEN,
```
Comment (ekremney) [inline packages/spacecat-shared-google-client/src/index.js:26]: are access token and refresh token expected to come from env variables? 🤔 these should be short lived creds, unless we don't have a way to automatically rotate them, I guess we need to obtain these tokens freshly from google, not from env variables

Comment (dzehnder) [reply packages/spacecat-shared-google-client/src/index.js:26]: the idea was to write a customer secret wrapper, that adds the secrets as env variables for the duration of the function execution.
Do you think this should be handled differently?

Comment (ekremney) [reply packages/spacecat-shared-google-client/src/index.js:26]: > the idea was to write a customer secret wrapper, that adds the secrets as env variables for the duration of the function execution.

where the "customer secret wrapper" is expected to read those secrets from? request body or the query params? If that's the case, then we can get it from `context.data` or `context.params`

**Comment by solaris007** [comment]
some improvements discussed f2f (checkauth method and input validation, version pinning)

**Review by solaris007 — APPROVED** [review]
LGTM

**Pull request merged**